java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -742,15 +742,22 @@ private JavaSymbol selectMostSpecific(JavaSymbol m1, JavaSymbol m2, List<JavaTyp
       return m1;
     }
     TypeSubstitution m2Substitution = null;
-    if (((JavaSymbol.MethodJavaSymbol) m2).isParametrized()) {
+    boolean m1IsGeneric = ((JavaSymbol.MethodJavaSymbol) m1).isParametrized();
+    boolean m2IsGeneric = ((JavaSymbol.MethodJavaSymbol) m2).isParametrized();
+    if (m2IsGeneric) {
       m2Substitution = typeSubstitutionSolver.getTypeSubstitution((JavaSymbol.MethodJavaSymbol) m2, callSite, ImmutableList.of(), argTypes);
     }
     if (m2Substitution == null) {
       m2Substitution = new TypeSubstitution();
     }
     boolean m1SignatureMoreSpecific = isSignatureMoreSpecific(m1, m2, argTypes, m1Substitution, m2Substitution);
-    boolean m2SignatureMoreSpecific = isSignatureMoreSpecific(m2, m1, argTypes, m1Substitution, m2Substitution);
+    boolean m2SignatureMoreSpecific = isSignatureMoreSpecific(m2, m1, argTypes, m2Substitution, m1Substitution);
     if (m1SignatureMoreSpecific && m2SignatureMoreSpecific) {
+      if(!m1IsGeneric && m2IsGeneric) {
```

Comment:
This deserves a reference to JLS (I'm not sure it fits https://docs.oracle.com/javase/specs/jls/se10/html/jls-18.html#jls-18.5.4 ?)

From my point of view, this is also maybe incomplete, as the JLS speaks about type inference, which is not necessarily the case with all generics.